### ISSUE TITLE
(ha)ploidy

### ISSUE THREAD

**Issue by Perugolate** [lh3/fermikit#3]
is it possible to change the ploidy for the variant calling steps (can't see a flag)?

**Comment by lh3** [comment]
The entire de novo assembly process makes no assumptions on ploidy, I believe. Variant calling does assume a diploid sample, but it should not be too hard to drop this assumption (EDIT: hmm.. still need some code refactoring). I will leave this issue open.

**Comment by lh3** [comment]
For haploid genomes, I would recommend to call them as diploid and then filter out (spurious) heterozygotes.

**Comment by ekg** [comment]
What about pooled samples, perhaps with tens or hundreds of individual
genomes represented?

**Comment by lh3** [comment]
BFC keeps a base if there are 4 or more base of the same type. Unitig construction is not directly affected by ploidy or base frequency (though low-frequency haplotypes are harder to assemble). Graph simplification may pop a bubble when the frequency is below 15% (controlled by option `fermi2 simplify -r`). Variant calling is unaware of ploidy. Filtering sets a threshold on allele balance, so it will filter low-frequency alleles.

In summary, fermikit is tuned for diploid samples, but with minor modifications, it should work with pooled samples or samples having low-frequency alleles.

**Comment by ekg** [comment]
That's excellent to hear. I have been wondering if we could build graph
reference systems using fermikit on top of pooled samples. It would seem
that this should work, barring memory issues caused by increasing the input
read set.

Is there any way to merge cleaned graphs from different single-sample
assemblies?

Perhaps this is a topic to discuss off-thread.

**Comment by lh3** [comment]
You can assemble unitigs with:

``` sh
cat *.mag.gz | ropebwt2 -dr > unitigs.fmd
fermi2 assemble -l51 -t16 unitigs.fmd > assembled.mag
```

`simplify` would not work well. you will also lose per-base read depth.

**Comment by ekg** [comment]
Losing the per-base depth is not a problem, as the idea will be to realign
the input sequencing reads to the graph. This will provide full annotation
of sequencing quality and depth for anything in the graph. Also, pruning
bubbles created by small variants would not be too much of a problem as we
can recover these via resequencing against the combined graph.
